python/embeddings.py

```diff
@@ -0,0 +1,139 @@
+import os
+import sys
+import multiprocessing as mp
+from concurrent.futures import ProcessPoolExecutor, as_completed
+import mgp
+
+sys.path.append(os.path.join(os.path.dirname(__file__), "embed_worker"))
+
+logger: mgp.Logger = mgp.Logger()
+
+EXCLUDE_PROPERTIES = {"embedding"}
+BATCH_SIZE = 2000
+MODEL_ID = "all-MiniLM-L6-v2"
+
+
+def build_texts(vertices):
+    out = []
+    for v in vertices:
+        txt = " ".join(lbl.name for lbl in v.labels) + " " + " ".join(
+            f"{k}: {val}" for k, val in v.properties.items() if k not in EXCLUDE_PROPERTIES
+        )
+        out.append(txt)
+    return out
+
+
+def split_slices(n_items: int, n_parts: int):
+    base, rem = divmod(n_items, n_parts)
+    start = 0
+    slices = []
+    for i in range(n_parts):
+        end = start + base + (1 if i < rem else 0)
+        slices.append((start, end))
+        start = end
+    return slices
+
+
+def get_visible_gpus():
+    # Avoid creating a CUDA context in the parent if possible
+    try:
+        import subprocess
+        out = subprocess.check_output(
+            ["nvidia-smi", "--query-gpu=index", "--format=csv,noheader"], text=True
+        )
+        return [int(x) for x in out.strip().splitlines() if x.strip()]
+    except Exception:
+        try:
+            import torch
+            return list(range(torch.cuda.device_count())) if torch.cuda.is_available() else []
+        except Exception:
+            return []
+
+
+@mgp.write_proc
+def compute_embeddings(ctx: mgp.ProcCtx, input_vertices: mgp.Nullable[mgp.List[mgp.Vertex]] = None) -> mgp.Record(success=bool):
+    logger.info(
+        f"compute_embeddings: starting (py_exec={sys.executable}, py_ver={sys.version.split()[0]})"
+    )
+    try:
+        # Parent imports are okay; workers import only embed_worker
+        import embed_worker  # <-- our pure worker module
+    except Exception as e:
+        logger.error(f"Failed to import worker module: {e}")
+        return mgp.Record(success=False)
+
+    try:
+        if input_vertices:
+            vertices = input_vertices
+        else:
+            vertices = list(ctx.graph.vertices)
```

Comment:
Batching is also required here because this will pull all the vertex accessor into the QM memory